search-app/src/cmr/search/api/autocomplete.clj

```diff
@@ -0,0 +1,63 @@
+(ns cmr.search.api.autocomplete
+  "Defines the API for autocomplete-suggest in the CMR."
+  (:require
+    [clojure.string :as str]
+    [cmr.common.log :refer (debug info warn error)]
+    [cmr.common.services.errors :as svc-errors]
+    [cmr.search.services.autocomplete-service :as ac]
+    [cmr.search.services.query-service :as qs]
+    [cmr.common-app.api.routes :as common-routes]
+    [cmr.common.mime-types :as mt]
+    [compojure.core :refer :all]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Support Functions
+
+(def limit-default 20)
+(def offset-default 0)
+
+(defn- string-param-or-default->int
+  [s default]
+  (if s
+    (Integer/parseInt s 10)
+    default))
+
+(defn- get-autocomplete-suggestions
+  "Invokes Elasticsearch for autocomplete query result"
+  [ctx params]
+  (let [query-param (:q params)
+        type-param (:types params)]
+    (if query-param
+      (let [term (str/trim (str/lower-case query-param))
+            types (if type-param
+                    (drop-while empty?
```

Comment:
Don't think `drop-while` is the right function to use here. I think you want to continue until the end of the list, so consider using `filter` or `keep`. Also combining lower-case and trim into one function, so you don't have to process the list two times.